R/aggregate_by_matrix.R

```diff
@@ -0,0 +1,37 @@
+#' @rdname aggregate
+#'
+#' @export
+colMeansMat <- function(x, MAT)
+  colSumsMat(x, MAT)/colSums(MAT)
+
+#' @rdname aggregate
+#'
+#' @export
+colSumsMat <- function(x, MAT)
+  t(MAT) %*% x
```

Comment:
(maybe neglectable speed improvement); from `?crossprod`:
> This is formally equivalent to (but usually
     slightly faster than) the call ‘t(x) %*% y’ (‘crossprod’) or ‘x
     %*% t(y)’ (‘tcrossprod’).

```suggestion
  crossprod(MAT, x)
```